glass/core/array.py

```diff
@@ -72,10 +101,19 @@ def trapz_product(f, *ff, axis=-1):  # type: ignore[no-untyped-def]
     y = np.interp(x, *f)
     for f_ in ff:
         y *= np.interp(x, *f_)
-    return np.trapz(y, x, axis=axis)  # type: ignore[attr-defined]
+    return np.trapz(  # type: ignore[attr-defined]
+        y,
+        x,
+        axis=axis,
+    )
 
 
-def cumtrapz(f, x, dtype=None, out=None):  # type: ignore[no-untyped-def]
+def cumtrapz(
+    f: npt.NDArray[np.int_] | npt.NDArray[np.float64],
+    x: npt.NDArray[np.int_] | npt.NDArray[np.float64],
+    dtype: type | None = None,
```

Comment:
better to use [DTypeLike](https://numpy.org/doc/stable/reference/typing.html#numpy.typing.DTypeLike)